### DISCUSSION TITLE
Support for Google Nest Mini mod hardware

### DISCUSSION THREAD

**Discussion by justLV** [toverainc/willow#237] [Ideas]
I just saw this project now and it looks great - I built a similar (albeit more of a demo) project using a Google Nest Mini, see [demo](https://twitter.com/justLV/status/1681377298308820992?s=20).

I'm happy to provide support and early sample hardware for integration if this is a more interesting form factor to the ESP32 Box - I'm agnostic to the inference server used, just feel this concept needs to get out there!

Crazy (but not that unexpected) coincidence we selected for the same chipset.

Code is over [here](https://github.com/justLV/onju-voice), w. link to the schematics. Will this be portable as is or are there any architecture changes needed?

cc @kristiankielhofner

**Comment by kristiankielhofner** [comment]
This is a very interesting project, nice work!

As you've probably seen we're an ESP-IDF based project (for the reasons you note and more).

We could potentially support your ESP32-S3 based PCB but I'm curious - what is the estimated end-user cost?

**Comment by justLV** [comment]
Sounds good.

The cost is TBD on volume, assembly costs, tariffs, shipping & Crowd Supply margin, but I'd estimate $30-50. 

The Google Nest Mini's I've been picking up at ~$20 off eBay, $38 from Walmart for a new one plus some bundle discounts. Alternatively, a 3D printed base, $6 shell off of Aliexpress, and $2 speaker could also be a form factor that needs a little more design attention.

The schematics are up [here](https://github.com/justLV/onju-voice/blob/master/images/Schematic.pdf) - let me know if this would be doable w/o any changes.

Personally I'd love to test it out, and happy to send you a unit or two to do the same. (I just received a ESP32-Box I ordered and TBH the build quality looks pretty janky!)

**Reply by kristiankielhofner** [reply]
From a quick review of the schematic integration to Willow could be challenging. I'm not sure if IDF/ADF will easily support your microphones and speaker driver. I haven't done much research on this with IDF/ADF so it could be straightforward, I don't know. We're pretty busy on this side; if you could take a look at this that would be much appreciated!

The ESP BOX is a great off the shelf device for Willow, especially considering the "$50, flash, and install" price point. Espressif is aware the build quality leaves a bit to be desired and they are working on an improved enclosure for the next revision. That said it would be nice to support another device (even via a Google Nest Mini shell).

**Reply by justLV** [reply]
No drivers should be necessary given a codec chip doesn't need to be setup, and you should be able to just use the [I2S stream](https://docs.espressif.com/projects/esp-adf/en/v2.0/api-reference/streams/index.html#i2s-stream), but will let the choice of integrating it be up to you / your community, no pressure from my side.

**Reply by kristiankielhofner** [reply]
It's been a while since I looked at this code but we don't really read from i2s directly, we use the AFE from ESP-SR and to my knowledge it requires drivers for the codec chip.
